library/std/src/io/mod.rs

```diff
@@ -2777,22 +2777,49 @@ pub struct Bytes<R> {
 impl<R: Read> Iterator for Bytes<R> {
     type Item = Result<u8>;
 
-    #[inline]
     fn next(&mut self) -> Option<Result<u8>> {
+        SpecReadByte::spec_read_byte(&mut self.inner)
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        SizeHint::size_hint(&self.inner)
+    }
+}
+
+trait SpecReadByte {
```

Comment:
This could use a one-liner explaining what it does. The naming isn't idea. If the methods were named "slow path" and "fast path" it might be clearer but since this mixes fast/slow path and specialization someone who just stumbles on this might be a bit confused.

Reply:
:+1: